src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,139 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int index;
+    private Node<K, V>[] storage;
+    private Node<K, V> temp;
+
+    public MyHashMap() {
+        storage = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkLoadedCapacity();
+        if (key == null) {
+            putByIndex(null, value, 0);
+        } else {
+            index = definePosition(key);
+            putByIndex(key, value, index);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size > 0) {
+            index = definePosition(key);
+            temp = storage[index];
+            while (temp != null) {
+                if (Objects.equals(key, temp.key)) {
+                    return temp.value;
+                }
+                temp = temp.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public V remove(K key) {
+        index = definePosition(key);
+        temp = storage[index];
+        V result = null;
+        if (Objects.equals(storage[index].key, key)) {
+            result = storage[index].value;
+            storage[index] = storage[index].next;
+        } else {
+            while (storage[index].next != null) {
+                if (Objects.equals(temp.next.key, key)) {
+                    result = temp.next.value;
+                    temp.next = temp.next.next;
+                    break;
+                }
+
+                temp = temp.next;
+            }
+        }
+        size -= (result == null) ? 0 : 1;
+        return result;
+    }
+
+    private Node<K, V>[] resizeArray() {
+        Node<K, V>[] boxes = new Node[storage.length];
+        System.arraycopy(storage, 0, boxes, 0, storage.length);
+        storage = new Node[storage.length * 2];
```

Comment:
The `resizeArray` method creates a new array `boxes` with the same length as `storage`, but it should initialize `storage` with double the capacity to accommodate more elements. Modify the initialization of `storage` to `new Node[storage.length * 2]`.